cni/pkg/iptables/iptables.go

```diff
@@ -156,14 +156,14 @@ func (cfg *IptablesConfigurator) executeDeleteCommands() error {
 	deleteCmds := [][]string{
 		{"-t", iptablesconstants.MANGLE, "-D", iptablesconstants.PREROUTING, "-j", ChainInpodPrerouting},
 		{"-t", iptablesconstants.MANGLE, "-D", iptablesconstants.OUTPUT, "-j", ChainInpodOutput},
-		{"-t", iptablesconstants.NAT, "-D", iptablesconstants.PREROUTING, "-j", ChainInpodPrerouting},
 		{"-t", iptablesconstants.NAT, "-D", iptablesconstants.OUTPUT, "-j", ChainInpodOutput},
-		{"-t", iptablesconstants.RAW, "-D", iptablesconstants.PREROUTING, "-j", ChainInpodPrerouting},
```

Comment:
While working on the initial batch of iptables tests, I realized that the delete functions expect certain commands to work despite the relevant rules not always being present (as they are not needed for tproxy or ingress mode).

To fix this, I moved certain rules into the optional delete commands.
I'm not sure if the old behavior was intentional or if simply no user raised an issue about it yet :confused:

Reply:
>  I'm not sure if the old behavior was intentional or if simply no user raised an issue about it yet 😕

The latter :D - thanks!

Reply:
Good to know :)
I have added a small mention in the release notes about this